docs/proposals/horizontal-pod-autoscaler.md

```diff
@@ -0,0 +1,272 @@
+(there will be API on master for this purpose, see
+[#11951](https://github.com/GoogleCloudPlatform/kubernetes/issues/11951).
+Then, it will compare the current CPU or memory consumption with the Target,
+and adjust the count of the Scale if needed to match the target
+(preserving condition: MinCount <= Count <= MaxCount).
+
+The target number of pods will be calculated from the following formula:
+
+```
+TargetNumOfPods = sum(CurrentPodsConsumption) / Target
+```
+
+To make scaling more stable, scale-up will happen only when the floor of ```TargetNumOfPods``` is higher than
+the current number, while scale-down will happen only when the ceiling of ```TargetNumOfPods``` is lower than
+the current number.
+
+The decision to scale-up will be executed instantly.
+However, we will execute scale-down only if the sufficient time has past from the last scale-up (e.g.: 10 minutes).
+Such approach has two benefits:
+
+* Autoscaler works in a conservative way.
+  If new user load appears, it is important for us to rapidly increase the number of pods,
+  so that user requests will not be rejected.
+  Lowering the number of pods is not that urgent.
+
+* Autoscaler avoids thrashing, i.e.: prevents rapid execution of conflicting decision if the load is not stable.
+
+
+As the CPU consumption of a pod immediately after start may be highly variable due to initialization/startup,
+autoscaler will skip metrics from the first minute of pod lifecycle.
+
+## Relative vs. absolute metrics
+
+The question arises whether the values of the target metrics should be absolute (e.g.: 0.6 core, 100MB of RAM)
+or relative (e.g.: 110% of resource request, 90% of resource limit).
+The argument for the relative metrics is that when user changes resources for a pod,
+she will not have to change the definition of the autoscaler object, as the relative metric will still be valid.
+However, we want to be able to base autoscaling on custom metrics in the future.
+Such metrics will rather be absolute (e.g.: the number of queries-per-second).
+Therefore, we decided to give absolute values for the target metrics in the initial version.
+
+Please note that when custom metrics are supported, it will be possible to create additional metrics
+in heapster that will divide CPU/memory consumption by resource request/limit.
+From autoscaler point of view the metrics will be absolute,
+althoug such metrics will be bring the benefits of relative metrics to the user.
+
+
+## Support in kubectl
+
+To make manipulation on HorizontalPodAutoscaler object simpler, we will add support for creating/updating/deletion/listing of HorizontalPodAutoscaler to kubectl.
+In addition, we will add kubectl support for the following use-cases:
+* When running an image with ```kubectl run```, there should be an additional option to create
+  an autoscaler for it.
+* When creating a replication controller or deployment with ```kubectl create [-f]```, there should be
```

Comment:
This is automatically true since you can include multiple objects in the same config file, right?